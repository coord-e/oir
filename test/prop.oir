# test: "$OIR_BIN" -p reach,prop
LABEL entry
  r1 = IMM 10
  r2 = MOV r1
  r3 = ADD r1 r2
  BRANCH r2 a b
LABEL a
  r4 = IMM 1
  JUMP end
LABEL b
  r4 = IMM 2
  JUMP end
LABEL end
  r5 = MOV r4     # should not be propagated
  r6 = IMM 20
  r7 = ADD r3 r6  # should be propagated
  RETURN r7
