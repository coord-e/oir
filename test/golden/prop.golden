# OIR entry=0, exit=3
# BB0 succs={1, 2} preds={}
LABEL entry  # 0
  r1 = IMM 10
  r2 = IMM 10
  r3 = IMM 20
  JUMP a  # 1
# BB1 succs={3} preds={0}
LABEL a  # 1
  r4 = IMM 1
  JUMP end  # 3
# BB2 succs={3} preds={0}
LABEL b  # 2
  r4 = IMM 2
  JUMP end  # 3
# BB3 succs={} preds={1, 2}
LABEL end  # 3
  r5 = MOV r4
  r6 = IMM 20
  r7 = IMM 40
  RETURN r7
